Permanent/ViewControllers/TagDetailsViewController.swift

```diff
@@ -0,0 +1,260 @@
+    
+    @IBOutlet weak var tagFindSearchBar: UISearchBar!
+    @IBOutlet weak var AddTagButton: RoundedButton!
+    @IBOutlet weak var tagsCollectionView: UICollectionView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        styleNavBar()
+        initUI()
+
+        tagsCollectionView.register(UINib(nibName: TagCollectionViewCell.identifier, bundle: nil), forCellWithReuseIdentifier: TagCollectionViewCell.identifier)
+        
+        self.tagsCollectionView.dataSource = self
+        self.tagsCollectionView.delegate = self
+        self.tagFindSearchBar.delegate = self
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        view.endEditing(true)
+    }
+    
+    override func styleNavBar() {
+        super.styleNavBar()
+        
+        navigationController?.navigationBar.barTintColor = .black
+    }
+    
+    func initUI() {
+        view.backgroundColor = .black
+        
+        navigationItem.title = "Edit Tags".localized()
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancelButtonPressed(_:)))
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(doneButtonPressed(_:)))
+        
+        tagFindSearchBar.setDefaultStyle(placeholder: "Add new tag".localized())
+        tagFindSearchBar.setFont(Text.style2.font)
+        tagFindSearchBar.setPlaceholderTextColor(.lightGray)
+        tagFindSearchBar.setTextColor(.lightGray)
+        tagFindSearchBar.setBackgroundColor(.darkGray)
+        tagFindSearchBar.tintColor = .lightGray
+        tagFindSearchBar.barTintColor = .lightGray
+        tagFindSearchBar.autocapitalizationType = .none
+        
+        AddTagButton.configureActionButtonUI(title: "Add".localized(), bgColor: .barneyPurple, buttonHeight: CGFloat(30))
+        
+        tagsCollectionView.backgroundColor = .clear
+        tagsCollectionView.indicatorStyle = .white
+        
+        initTagsState()
+    
+        let columnLayout = CustomViewFlowLayout()
+        tagsCollectionView.collectionViewLayout = columnLayout
+    }
+    
+    @objc func cancelButtonPressed(_ sender: UIBarButtonItem) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc func doneButtonPressed(_ sender: UIBarButtonItem) {
+        let dispatchGroup = DispatchGroup()
+        showSpinner(colored: .lightGray)
+        
+        let forAddingBool = sortedArray.map({ (SortedTagVO) in
+            return SortedTagVO.forAdding
+         })
+        let forAddingNames = sortedArray.map { (item) -> String in
+            return item.tagVO.tagVO.name ?? ""
+        }
+        
+        let addedNames: [String] = zip(forAddingBool, forAddingNames).filter{ $0.0 }.map{ $1 }
```

Comment:
This whole logic, starting from line 102 to 109 can be simplified to:

let addedTags: [String] = sortedArray.filter({ $0.forAdding }).map({ $0.tagVO.tagVO.name ?? "" })

Reply:
Logic was replaced.